org.eclipse.winery.workflowmodeler/src/app/model/workflow.node.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2017 ZTE Corporation.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v10.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
+ *     ZTE - initial API and implementation and/or initial documentation
+ */
+
+export const WorkflowNodeType = [
+    'StartEvent',
+    'EndEvent',
+    'ToscaNodeManagementTask',
+    'ExclusiveGateway',
+    'ParallelGateway',
+];
+
+export class WorkflowNode {
+    public id: string;
+    public name: string;
+    public type: string;
+    public input: any[];
+    public output: any[];
+    public connection: any[];
+    public template: any;
+    public position: any;
+
+    public nodeInterface: string;
+    public nodeTemplate: string;
+    public nodeOperation: string;
+
+    constructor({

```

Comment:
Yes, but, in my opinion however, there would be a cleaner and much more readable approach, if you would use a real class here instead of declaring the object here. So you could have something like

```
export class WorkflowNode {
    constructor (public data: WorflowNodeData) {
    }
}

export class WorflowNodeData {
        id: sting;
        name: sting;
        type: string;
        position = {};
        input = [];
        connection = [];
        output = [];
        template;
        interface: nodeInterface;
        node_operation: nodeOperation;
        node_template: nodeTemplate;
}
```

With this approach, you would have a better separation of concerns and a more consistent and more readable code. As I said, in my opinion. Mybe @koppor can tell us his opinion.